es/ae/oracle.js

```diff
@@ -61,27 +62,23 @@ async function getOracleObject (oracleId) {
  * @alias module:@aeternity/aepp-sdk/es/ae/oracle
  * @instance
  * @function
- * @category async
  * @param {String} oracleId Oracle public key
  * @param {Function} onQuery OnQuery callback
  * @param {Object} [options] Options object
  * @param {Number} [options.interval] Poll interval(default: 5000)
  * @return {Function} stopPolling - Stop polling function
  */
-async function pollForQueries (oracleId, onQuery, { interval = 5000 } = {}) {
-  const queries = (await this.getOracleQueries(oracleId)).oracleQueries || []
-  let quriesIds = queries.map(q => q.id)
-  await onQuery(queries)
-
-  async function pollQueries () {
-    const { oracleQueries: q } = await this.getOracleQueries(oracleId)
-    const newQueries = q.filter(({ id }) => !quriesIds.includes(id))
-    if (newQueries.length) {
-      quriesIds = [...quriesIds, ...newQueries.map(a => a.id)]
-      onQuery(newQueries)
-    }
+function pollForQueries (oracleId, onQuery, { interval = 5000 } = {}) {
+  const sentQueryIds = new Set()
+  const fetchQueries = async () => {
+    const queries = ((await this.getOracleQueries(oracleId)).oracleQueries || [])
+      .filter(({ id }) => !sentQueryIds.has(id))
+    queries.forEach(({ id }) => sentQueryIds.add(id))
```

Comment:
I'd refactor this to use map again.

(not tested)
```javascript
const allQueries = (await this.getOracleQueries(oracleId)).oracleQueries || [])
const queries = [...new Set(allQueries.map(({ id }) => id))]
```

Reply:
It looks like we don't need to store all previous queries, just the last one instead, and pass it as `from` parameter to GetOracleQueriesByPubkey. I would like to check it later, proposing to leave as it is now.
https://github.com/aeternity/aepp-sdk-js/issues/1055